scripts/scripts.js

```diff
@@ -90,56 +90,62 @@ function openInSameTab(url) {
   );
 }
 
-function decorateTIAAButtons(main) {
-  main.querySelectorAll('a').forEach((a) => {
-    const openInNewTab = openInSameTab(a.href);
+function addIconForExternalLinks(a) {
+  const openInNewTab = openInSameTab(a.href);
 
-    const cta = domEl(
-      'qui-wc-cta',
-      {
-        'cta-type':
+  const cta = domEl(
+    'qui-wc-cta',
+    {
+      'cta-type':
         'link',
-        href: a.href,
-        size: 'medium',
-        class: 'qui-cta',
-      },
-    );
-
-    if (!openInNewTab) {
-      a.setAttribute('target', '_blank');
-      cta.setAttribute('target', '_blank');
-      if (!isPdf(a.href)) {
-        a.title += ' Opens in a new tab';
-        a.appendChild(span({ class: 'icon icon-ethos-launch' }));
-      } else {
-        a.title += ' Opens PDF';
-        a.appendChild(span({ class: 'icon icon-ethos-document_outline' }));
-      }
+      href: a.href,
+      size: 'medium',
+      class: 'qui-cta',
+    },
+  );
+
+  if (!openInNewTab) {
+    a.setAttribute('target', '_blank');
+    cta.setAttribute('target', '_blank');
+    if (!isPdf(a.href)) {
+      a.title += ' Opens in a new tab';
+      a.appendChild(span({ class: 'icon icon-ethos-launch' }));
+    } else {
+      a.title += ' Opens PDF';
+      a.appendChild(span({ class: 'icon icon-ethos-document_outline' }));
     }
+  }
+
+  a.setAttribute('aria-label', a.title);
+  cta.setAttribute('qui-aria-label', a.title);
 
-    a.setAttribute('aria-label', a.title);
-    cta.setAttribute('qui-aria-label', a.title);
-
-    if (a.classList.contains('button')) {
-      cta.setAttribute('cta-appearance', 'button');
-      a.setAttribute('cta-appearance', 'button');
-      a.setAttribute('mat-button', true);
-
-      const aClass = ['mat-flat-button', 'mat-primary'];
-      if (a.classList.contains('primary')) {
-        cta.setAttribute('variant', 'flat');
-      } else {
-        cta.setAttribute('variant', 'stroke');
-        aClass.push('mat-stroked-button');
-      }
-      a.className = aClass.join(' ');
+  if (a.classList.contains('button')) {
+    cta.setAttribute('cta-appearance', 'button');
+    a.setAttribute('cta-appearance', 'button');
+    a.setAttribute('mat-button', true);
+
+    const aClass = ['mat-flat-button', 'mat-primary'];
+    if (a.classList.contains('primary')) {
+      cta.setAttribute('variant', 'flat');
     } else {
-      a.setAttribute('cta-appearance', 'link');
-      cta.setAttribute('cta-appearance', 'link');
+      cta.setAttribute('variant', 'stroke');
+      aClass.push('mat-stroked-button');
     }
+    a.className = aClass.join(' ');
+  } else {
+    a.setAttribute('cta-appearance', 'link');
+    cta.setAttribute('cta-appearance', 'link');
+  }
+
+  a.replaceWith(cta);
+  cta.appendChild(a);
+}
 
-    a.replaceWith(cta);
-    cta.appendChild(a);
+function decorateTIAAButtons(main) {
+  main.querySelectorAll('a').forEach((a) => {
+    if (a.getElementsByClassName('icon').length === 0) {
+      addIconForExternalLinks(a);
```

Comment:
I don't fully understand this part. The previous function was doing way more than adding an icon for external links, I was doing the translation from the Edge Delivery semantic of links/buttons -> TIAA semantic of links/buttons. Adding the icons was just a small part of that.

In these two lines:
https://github.com/aemsites/tiaa/pull/22/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cR112

https://github.com/aemsites/tiaa/pull/22/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cR115

My understanding was that we would check in those two specific places before appending icons, not completely deactivating the logic, if an icon is present.

Reply:
done